spec/ysh-stdlib-args.test.sh

```diff
@@ -9,7 +9,7 @@
 
 source --builtin args.ysh
 
-Args (&spec) {
+arg-parse (&spec) {
```

Comment:
We don't have to resolve this now, but it's hard to find a good name for this

I would think of it as noun-vs-verb

I think the "spec" should be a noun.  Though I've come to dislike "spec" because it seems like a wishy-washy and overused word

'arg-parse' seems like a verb.  `arg-parser` would be closer

But we can think about it